### ISSUE TITLE
StackOverflowError caused by non-compiling superclass recursion in source code

### ISSUE THREAD

**Issue by Firestar99** [stokito/IdeaJol#29]
StackOverflowError occurs when typing source code and a "superclass recursion" is typed, see example below. 
```
class Buffer extends Buffer {

}
```
Of course very rare to actually happen, but you may want to create an (optionally inner) class Buffer and have it extend my.project.Buffer by first typing Buffer and then autocompleting the fully qualified name, in that short moment before autocompleting the IDE complains about the Plugin crashing. It's just reporting that the plugin crashed and you experience a short lag spike but otherwise does not disrupt your work much if at all.

Thought I'd still report it even if the impact is very little :) Like your plugin!

```
java.lang.StackOverflowError
	at com.intellij.psi.impl.source.PsiFileImpl.getStubTree(PsiFileImpl.java:618)
	at com.intellij.psi.impl.source.PsiFileImpl.getGreenStubTree(PsiFileImpl.java:952)
	at com.intellij.psi.impl.source.SpineRef.getGreenStub(SpineRef.java:39)
	at com.intellij.extapi.psi.StubBasedPsiElementBase.getGreenStub(StubBasedPsiElementBase.java:352)
	at com.intellij.psi.impl.source.PsiClassImpl.isAnnotationType(PsiClassImpl.java:426)
	at com.intellij.psi.impl.PsiClassImplUtil.getExtendsListTypes(PsiClassImplUtil.java:937)
	at com.intellij.psi.impl.source.PsiClassImpl.getExtendsListTypes(PsiClassImpl.java:185)
	at com.intellij.psi.impl.PsiClassImplUtil.getSuperClass(PsiClassImplUtil.java:711)
	at com.intellij.psi.impl.source.PsiClassImpl.getSuperClass(PsiClassImpl.java:200)
	at org.jetbrains.uast.UClass$DefaultImpls.getSuperClass(UClass.kt:36)
	at org.jetbrains.uast.java.AbstractJavaUClass.getSuperClass(JavaUClass.kt:9)
	at org.jetbrains.uast.java.JavaUClass.getSuperClass(JavaUClass.kt:51)
	at org.jetbrains.uast.java.JavaUClass.getSuperClass(JavaUClass.kt:46)
	at com.github.stokito.IdeaJol.PsiClassAdapter.createClassDataFromPsiClass(PsiClassAdapter.java:19)
	at com.github.stokito.IdeaJol.PsiClassAdapter.createClassDataFromPsiClass(PsiClassAdapter.java:20)
	at com.github.stokito.IdeaJol.PsiClassAdapter.createClassDataFromPsiClass(PsiClassAdapter.java:20)
	at com.github.stokito.IdeaJol.PsiClassAdapter.createClassDataFromPsiClass(PsiClassAdapter.java:20)
	[...]
```

**Comment by stokito** [comment]
I fixed, please retest. The "Cyclic inheritance involving" is an error but during editing this may occur.
Thank you for the detailed bug report

**Comment by Firestar99** [comment]
Bug is confirmed fixed, but found another related one: #32 
With this one feel free to ignore, see description. :)
